packages/streamr-logger/include/streamr-logger/Logger.hpp

```diff
@@ -1,18 +1,152 @@
-#ifndef STREAMER_UTILS_LOGGER_LOGGER_HPP
-#define STREAMER_UTILS_LOGGER_LOGGER_HPP
+#ifndef STREAMER_UTILS_LOGGER_LOGGER_H
+#define STREAMER_UTILS_LOGGER_LOGGER_H
 
-#include <iostream>
+#include <folly/logging/LogConfig.h>
+#include <folly/logging/LogLevel.h>
+#include <folly/logging/LogMessage.h>
+#include <folly/logging/LogWriter.h>
+#include <folly/logging/LoggerDB.h>
+#include <folly/logging/xlog.h>
+#include "StreamrWriterFactory.hpp"
+#include "StreamrHandlerFactory.hpp"
+
+namespace streamr {
+namespace logger {
+
+namespace {
+
+static const auto envLogLevelName = "LOG_LEVEL";
+static const auto logLevelTraceText = "trace";
+static const auto logLevelDebugText = "debug";
+static const auto logLevelInfoText = "info";
+static const auto logLevelWarnText = "warn";
+static const auto logLevelErrorText = "error";
+static const auto logLevelFatalText = "fatal";
+
+enum class StreamrLogLevel { trace, debug, info, warn, error, fatal };
+
+const std::unordered_map<StreamrLogLevel, folly::LogLevel> fromStreamrToFollyLogLevelMap{
+    {StreamrLogLevel::trace, folly::LogLevel::DBG},
+    {StreamrLogLevel::debug, folly::LogLevel::DBG0},
+    {StreamrLogLevel::info, folly::LogLevel::INFO},
+    {StreamrLogLevel::warn, folly::LogLevel::WARN},
+    {StreamrLogLevel::error, folly::LogLevel::ERR},
+    {StreamrLogLevel::fatal, folly::LogLevel::CRITICAL}};
+
+const std::unordered_map<std::string, folly::LogLevel> toFollyLogLevelMap{
+    {logLevelTraceText, folly::LogLevel::DBG},
+    {logLevelDebugText, folly::LogLevel::DBG0},
+    {logLevelInfoText, folly::LogLevel::INFO},
+    {logLevelWarnText, folly::LogLevel::WARN},
+    {logLevelErrorText, folly::LogLevel::ERR},
+    {logLevelFatalText, folly::LogLevel::CRITICAL}};
+
+std::optional<folly::LogLevel> getFollyLogLevelFromEnv() {
+    char* val = getenv(envLogLevelName);
+    if (val) {
+        auto pair = toFollyLogLevelMap.find(val);
+        if (pair != toFollyLogLevelMap.end()) {
+            return std::optional<folly::LogLevel>{std::move(pair->second)};
+        }
+    }
+    return std::nullopt;
+}
+
+}; // namespace
 
-namespace streamr::logger {
 class Logger {
 public:
-    Logger() = default;
 
-    // NOLINTNEXTLINE(readability-convert-member-functions-to-static)
-    void log(const std::string& message) const {
-        std::cout << message << std::endl;
+    Logger(std::shared_ptr<folly::LogWriter> logWriter = nullptr)
```

Comment:
Make an overloaded version Logger() so it looks nicer in IDEs? 
Also support the parameters used in the typescript version. Maybe std::optional<something> could be also useful? Anyway, see what looks nicest in auto-complete